### ISSUE TITLE
[ENH] Use CUROCKET as GPU implementation for ROCKET

### ISSUE THREAD

**Issue by oleeven** [aeon-toolkit/aeon#3345]
### Describe the feature or idea you want to propose

I wrote a GPU-based implementation for ROCKET using CUDA. It is ~20x faster and ~11x more computationally efficient per watt than the CPU implementation using numba (tested on AMD EPYC 7443P and NVIDIA RTX 3090). There is a [paper](https://arxiv.org/abs/2601.17091) describing the implementation and a corresponding [repo](https://github.com/oleeven/CUROCKET).

### Describe your proposed solution

CUROCKET is efficient because it does not try to squeeze imhomogenous convolutional kernels into a framework like pytorch or tensorflow. Instead, it uses a custom CUDA kernel. It is capable of handling multivariate data and running on multiple GPUs in parallel by splitting the data. Therefore, i propose CUROCKET as an alternative to RocketGPU in aeon.

### Describe alternatives you've considered, if relevant

The obvious alternative is the RocketGPU implementation currently used in aeon. Tbh it ran super slow on my setup, way slower than the CPU implementation, which is why I compared CUROCKET to the CPU implementation. Based on my findings, CUROCKET is significantly faster than RocketGPU. However, the main disadvantage of CUROCKET is that it is currently only implemented in CUDA.  I'm confident it could be ported to ROCM, but I have no experience with it.

**Comment by TonyBagnall** [comment]
him thanks for this, we would welcome a contribution if you are interested in porting it in, I personally have no experience of CUDA, curious the GPU version is so slow too, have you any idea why?

**Comment by MatthewMiddlehurst** [comment]
Yea there seems to be issues with the implementation. May just be worth deprecating given the performance loss but if anyone or @hadifawaz1999 can figure out why this is happening could be useful.

We have had suggestions for CUDA/Cupy before, but the consensus was that it is not really something we want to have to maintain I believe (#3233, #3211).

**Comment by oleeven** [comment]
I compared my version to that of @Adityakushwaha2006 [here](https://github.com/Adityakushwaha2006/aeon/tree/feat/cupy-rocket-acceleration/aeon/transformations/collection/convolution_based/rocketGPU). I tested on Threadripper PRO 5975WX and RTX PRO 5000 Blackwell and got ~3x speedup of his version to CPU and ~20x speedup of my version to CPU.

I read the discussions in #3233 and #3211. Since I also use CuPy, also with a custom kernel, I think it won't make sense for me to implement it in aeon unless the consensus changes on maintaining such code.
